sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -120,7 +120,8 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
         } else {
           mapStartIndices(i + 1)
         }
-        PartialReducerPartitionSpec(reducerId, startMapIndex, endMapIndex)
+        val dataSize = startMapIndex.until(endMapIndex).map(mapPartitionSizes(_)).sum
```

Comment:
Would it make sense to make this size part of the return value from `splitSizeListByTargetSize`?
Or even make `splitSizeListByTargetSize` return partition specs directly, like the other utility func?

Reply:
I'd like to keep `splitSizeListByTargetSize` simple so that we can reuse it in more places, e.g. coalescing the shuffle partitions.